cdc/generation.hh

```diff
@@ -113,6 +114,22 @@ public:
     const std::vector<token_range_description>& entries() const;
 };
 
+class streams_version {
```

Comment:
Could you document this class a bit?
Specifically, what does the "version" refer to?

Reply:
I can. It is a version of the set of streams.

Reply:
How are these sets of streams versioned?
Who/what defines the versioning?

If we have `streams_version sv1, sv2`, when is the version of `sv2` greater than `sv1`?

Also, maybe `versioned_streams` would be a better name?

Reply:
You version them. Version == generation. I.e. a topology change creates a new version of the stream id set. 
Maybe `stream_topology_generation` is a better name. Feel free to suggest.

Reply:
I avoided the `topology` name though, to not confuse it with something actually used in the topology calculation. This is purely for query of CDC_DESC.

Reply:
Ok, so it's versioned by the generation's timestamp.

This is basically a CDC generation but with the information about mapping of streams to token ranges forgotten. I.e. what we give to the client in CDC_DESC for querying streams.

Reply:
Maybe `generation_stream_set`?

`streams_version` is fine, but a comment would be nice in any case.